### ISSUE TITLE
Module tun not found in OpenVZ kernel 2.6.32-042stab127.2

### ISSUE THREAD

**Issue by Mikaela** [yggdrasil-network/yggdrasil-go#573]
Yggdrasil stops being able to start on Debian 9 in OpenVZ with kernel 2.6.32… after upgrade to 0.3.10. Earlier versions didn't have issues.

```
Oct 10 20:50:24 relpda systemd[1]: [/etc/systemd/system/yggdrasil.service:11] Unknown lvalue 'CapabilityBoundSet' in section 'Service'
Oct 10 20:50:24 relpda systemd[1]: yggdrasil.service: Failed to set invocation ID on control group /system.slice/yggdrasil.service, ignoring: Oper
ation not permitted                                                                  
Oct 10 20:50:24 relpda systemd[1]: Starting yggdrasil...
Oct 10 20:50:24 relpda yggdrasil[8092]: modprobe: ERROR: ../libkmod/libkmod.c:514 lookup_builtin_file() could not open builtin file '/lib/modules/
2.6.32-042stab127.2/modules.builtin.bin'
Oct 10 20:50:24 relpda yggdrasil[8092]: modprobe: FATAL: Module tun not found in directory /lib/modules/2.6.32-042stab127.2
Oct 10 20:50:24 relpda systemd[1]: yggdrasil.service: Control process exited, code=exited status=1
Oct 10 20:50:24 relpda systemd[1]: Failed to start yggdrasil.
Oct 10 20:50:24 relpda systemd[1]: yggdrasil.service: Unit entered failed state.
Oct 10 20:50:24 relpda systemd[1]: yggdrasil.service: Failed with result 'exit-code'.
```

**Comment by Mikaela** [comment]
Arceliar suggested a workaround of removing the modprobe line as a workaround and that works.

https://github.com/yggdrasil-network/yggdrasil-go/blob/045a24d74ed06f6aa658a131f3980fb8e60938b5/contrib/systemd/yggdrasil.service#L12

**Comment by Arano-kai** [comment]
According to [ExecStart](https://www.freedesktop.org/software/systemd/man/systemd.service.html#ExecStart=) syntax, prefixes may combine in any order, but it also may depends on systemd version (not sure).
Please, try to override with:
```
ExecStartPre=
ExecStartPre=+-/sbin/modprobe tun
```
and describe result.

**Comment by Mikaela** [comment]
I don't understand what you are asking me to do.

**Comment by Arano-kai** [comment]
```
# systemctl edit yggdrasil.service
##This will open your preferred editor

[Service]
##Discard any previous ExecStartPre
ExecStartPre=
##Set new directive
ExecStartPre=+-/sbin/modprobe tun
##save&close

# systemctl daemon-reload
# systemctl restart yggdrasil.service
```
This will create `/etc/systemd/system/yggdrasil.service.d/override.conf` that overrides denoted directives without touching original unit. Current overriding also disable autocreation of missing config, but this is not a problem if you already have one.
Please do that procedure and tell, if it helps.
In case of success, you can delete this file after arrival of update with fixed yggdrasil.service.

**Comment by prurigro** [comment]
Would it be worth adding a test to see if `/dev/net/tun` exists before trying to load the module so upgrading doesn't break yggdrasil on systems with it built into the kernel? Something like `[[ -c /dev/net/tun ]] || modprobe tun`

**Comment by Arano-kai** [comment]
Unnecessary, if we allow to fall execution (the '-' prefix)

**Comment by neilalexander** [comment]
This should be fixed now as of v0.3.11.

**Comment by Mikaela** [comment]
Sorry, I was away from GitHub/email a lot and missed the things you wanted me to try, but I hope this did get resolved. I haven't checked if the upgrade overwrote the yggdrasil.service.
